### PR TITLE
PostgreSQL v11.3 & v12.0-beta1 with PostGIS v3.0.0-alpha2 based on Alpine Linux

### PR DESCRIPTION
**Pull request by tobwen**
Incomplete, since Debian is missing. But it's a good way to allow people to test PostGIS-dev.

### CODE REVIEW

**Comment by jalessio** [comment]
@tobwen I tried building these images and ran into a few issues:

1. It looks like you've reversed the v11 and v12 `FROM` images in each `Dockerfile` you added. In `11.3-3.0.0alpha2/alpine/Dockerfile` it has `FROM postgres:12-beta1-alpine` and in `12.0beta1-3.0.0alpha2/alpine/Dockerfile` it has `FROM postgres:11.3-alpine`.
1. In both cases I'm getting errors like this when I try to `docker build .`:

```
Sending build context to Docker daemon   7.68kB
Step 1/7 : FROM postgres:12-beta1-alpine
 ---> cbdae8604e42
Step 2/7 : MAINTAINER Régis Belson <me@regisbelson.fr>
 ---> Using cache
 ---> 07a64c2cf87b
Step 3/7 : ENV POSTGIS_VERSION 3.0.0alpha2
 ---> Using cache
 ---> 5921f28aa2e8
Step 4/7 : ENV POSTGIS_SHA256 a4d538b4418f8dbc561676d96d8a396bf90a86b87a1085f7dbb0e3ed31275c90
 ---> Using cache
 ---> a6498cb07ae8
Step 5/7 : RUN set -ex         && apk add --no-cache --virtual .fetch-deps         ca-certificates         openssl         tar         && wget -O postgis.tar.gz "https://github.com/postgis/postgis/archive/$POSTGIS_VERSION.tar.gz"     && echo "$POSTGIS_SHA256 *postgis.tar.gz" | sha256sum -c -     && mkdir -p /usr/src/postgis     && tar         --extract         --file postgis.tar.gz         --directory /usr/src/postgis         --strip-components 1     && rm postgis.tar.gz         && apk add --no-cache --virtual .build-deps         autoconf         automake         g++         json-c-dev         libtool         libxml2-dev         make         perl         && apk add --no-cache --virtual .build-deps-edge         --repository http://dl-cdn.alpinelinux.org/alpine/edge/testing         --repository http://dl-cdn.alpinelinux.org/alpine/edge/main         gdal-dev         geos-dev         proj4-dev         protobuf-c-dev     && cd /usr/src/postgis     && ./autogen.sh     && ./configure     && make     && make install     && apk add --no-cache --virtual .postgis-rundeps         json-c     && apk add --no-cache --virtual .postgis-rundeps-edge         --repository http://dl-cdn.alpinelinux.org/alpine/edge/testing         --repository http://dl-cdn.alpinelinux.org/alpine/edge/main         geos         gdal         proj4         protobuf-c     && cd /     && rm -rf /usr/src/postgis     && apk del .fetch-deps .build-deps .build-deps-edge
 ---> Running in ea304538e936
+ apk add --no-cache --virtual .fetch-deps ca-certificates openssl tar
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/community/x86_64/APKINDEX.tar.gz
(1/4) Installing ca-certificates (20190108-r0)
(2/4) Installing openssl (1.1.1b-r1)
(3/4) Installing tar (1.32-r0)
(4/4) Installing .fetch-deps (0)
Executing busybox-1.29.3-r10.trigger
Executing ca-certificates-20190108-r0.trigger
OK: 54 MiB in 35 packages
+ wget -O postgis.tar.gz https://github.com/postgis/postgis/archive/3.0.0alpha2.tar.gz
Connecting to github.com (192.30.255.113:443)
Connecting to codeload.github.com (192.30.255.120:443)
postgis.tar.gz       100% |********************************| 15.3M  0:00:00 ETA

+ echo 'a4d538b4418f8dbc561676d96d8a396bf90a86b87a1085f7dbb0e3ed31275c90 *postgis.tar.gz'
+ sha256sum -c -
postgis.tar.gz: OK
+ mkdir -p /usr/src/postgis
+ tar --extract --file postgis.tar.gz --directory /usr/src/postgis --strip-components 1
+ rm postgis.tar.gz
+ apk add --no-cache --virtual .build-deps autoconf automake g++ json-c-dev libtool libxml2-dev make perl
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/community/x86_64/APKINDEX.tar.gz
(1/25) Installing m4 (1.4.18-r1)
(2/25) Installing libbz2 (1.0.6-r6)
(3/25) Installing perl (5.26.3-r0)
(4/25) Installing autoconf (2.69-r2)
(5/25) Installing automake (1.16.1-r0)
(6/25) Installing binutils (2.31.1-r2)
(7/25) Installing gmp (6.1.2-r1)
(8/25) Installing isl (0.18-r0)
(9/25) Installing libgomp (8.3.0-r0)
(10/25) Installing libatomic (8.3.0-r0)
(11/25) Installing mpfr3 (3.1.5-r1)
(12/25) Installing mpc1 (1.0.3-r1)
(13/25) Installing gcc (8.3.0-r0)
(14/25) Installing musl-dev (1.1.20-r4)
(15/25) Installing libc-dev (0.7.1-r0)
(16/25) Installing g++ (8.3.0-r0)
(17/25) Installing json-c (0.13.1-r0)
(18/25) Installing pkgconf (1.6.0-r0)
(19/25) Installing json-c-dev (0.13.1-r0)
(20/25) Installing libltdl (2.4.6-r5)
(21/25) Installing libtool (2.4.6-r5)
(22/25) Installing zlib-dev (1.2.11-r1)
(23/25) Installing libxml2-dev (2.9.9-r1)
(24/25) Installing make (4.2.1-r2)
(25/25) Installing .build-deps (0)
Executing busybox-1.29.3-r10.trigger
OK: 255 MiB in 60 packages
+ apk add --no-cache --virtual .build-deps-edge --repository http://dl-cdn.alpinelinux.org/alpine/edge/testing --repository http://dl-cdn.alpinelinux.org/alpine/edge/main gdal-dev geos-dev proj4-dev protobuf-c-dev
fetch http://dl-cdn.alpinelinux.org/alpine/edge/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/edge/testing/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/community/x86_64/APKINDEX.tar.gz
ERROR: unsatisfiable constraints:
  proj4-dev (missing):
    required by: .build-deps-edge-0[proj4-dev]
```

File: 11.3-3.0.0alpha2/alpine/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,63 @@
+FROM postgres:12-beta1-alpine
```
Comment (jalessio) [inline 11.3-3.0.0alpha2/alpine/Dockerfile]: This is the wrong `FROM` image for this Postgres 11 `Dockerfile`.

File: 12.0beta1-3.0.0alpha2/alpine/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,63 @@
+FROM postgres:11.3-alpine
```
Comment (jalessio) [inline 12.0beta1-3.0.0alpha2/alpine/Dockerfile]: This is the wrong `FROM` image for this Postgres 12 `Dockerfile`.

**Comment by tobwen** [comment]
I am really sorry, I lost track of this. I saw you already did it on your own: https://github.com/appropriate/docker-postgis/pull/142 In advance to your own development, I'm closing this PR.

**Pull request closed without merging**